### PR TITLE
added global monitor for window.Galaxy

### PR DESCRIPTION
**Pull request by Nerdinacan**
Still waiting on another PR before I can release most of what I've been working on, but this is a little diagnostic tool I've been using that can exist without webpack 4. 

It's a simple proxy monitor that watches window variables. I've set it up to watch window.Galaxy, some of the other horrible globals defined in onload.js. Actual functionality of the site should be unchanged. All this does is spam the console when the appropriate messages are toggled-on.

I've also used webpack's Provide plugin to define "Galaxy" as the resulting proxy object in all the webpack bundled scripts. (See webpack config, Provide plugin). This means it effectively watches all changes to window.Galaxy or Galaxy from the legacy scripts, and is intended to be used as a diagnostic tool while we refactor.

**To get a list of currently tracked window properties:**
showMonitoredProperties()

**To toggle the console messages on or off:**
toggleMonitorMessages("Galaxy", true|false)

**To watch a new window variable:**
import { installMonitor } from "utils/installMonitor";
installMonitor("SomeGlobalVariable", [optionalFallbackValue])

Then in the chrome/firefox/whatever console:
toggleMonitorMessage("SomeGlobalVariable", true)

And behold the spam.

### CODE REVIEW

**Comment by Nerdinacan** [comment]
Hmmm, looks like the karma configs are screwy. Will update those too.

File: test/galaxy_selenium/navigates_galaxy.py [hunk 1]
```diff
@@ -141,6 +141,9 @@ def wait_length(self, wait_type):
 
     def sleep_for(self, wait_type):
         time.sleep(self.wait_length(wait_type))
+    
+    def sleep_for_exactly(self, duration):
+        time.sleep(duration)
```
Comment (jmchilton) [inline test/galaxy_selenium/navigates_galaxy.py]: Intentional? Seems unused. I'm fine if you are using it downstream though, but do you want to change the definition of sleep_for to use sleep_for_exactly - removes some duplication and then the entry point is being used in the current code also?

```python
    def sleep_for(self, wait_type):
        self.sleep_for_exactly(self.wait_length(wait_type))
```

I'd also prefer the name ``sleep_for_duration`` or ``sleep_for_seconds`` but the current name is fine if you'd rather not change it.

Comment (Nerdinacan) [reply test/galaxy_selenium/navigates_galaxy.py]: Ahh I use it periodically during busted test diagnosis. I can rename it if you like. sleep_for_seconds sounds fine to me.

**Comment by Nerdinacan** [comment]
This is happy now that the unit tests can deal with a javascript Proxy. Merge whenever you like.

**Comment by dannon** [comment]
@Nerdinacan Awesome, thanks, let me check it out now.

**Comment by dannon** [comment]
@Nerdinacan Looks like we crossed updates here.  I'd merged locally without checking back in the PR window, so I didn't see your last merge from dev here.  It was just a merge with no conflicts, and no rebase, right?

(looks like it, I have an idea for a fix that'll mark this as merged -- will ping offline)

**Comment by dannon** [comment]
Well.  Github didn't update this to reflect what we wanted, but it's all merged and in there.  The only bug is that #6936 should make sure to include this in the release notes.

**Pull request closed without merging**